taxi/views.py

```diff
@@ -61,16 +63,29 @@ class CarListView(LoginRequiredMixin, generic.ListView):
 class CarDetailView(LoginRequiredMixin, generic.DetailView):
     model = Car
 
+    def post(self, request: HttpRequest, *args, **kwargs):
+        car = self.get_object()
+        user = self.request.user
+
+        if user in car.drivers.all():
+            car.drivers.remove(user)
+        else:
+            car.drivers.add(user)
+
+        return HttpResponseRedirect(
+            reverse("taxi:car-detail", kwargs={"pk": car.pk})
+        )
```

Comment:
In the `post` method of `CarDetailView`, ensure that the user is authorized to modify the car's drivers list. Consider adding permission checks to prevent unauthorized users from altering the car's driver assignments.